src/GafferCycles/IECoreCyclesPreview/CameraAlgo.cpp

```diff
@@ -115,27 +122,22 @@ ccl::Camera *convertCommon( const IECoreScene::Camera *camera, const std::string
 		ccam->border.clamp();
 	}
 	
-	// Shutter TODO: Need to see if this is correct or not, cycles also has a shutter curve...
+	// Shutter TODO: Cycles also has a shutter curve...
 	const Imath::V2f &shutter = camera->getShutter();
-	if ((shutter.x > 0.0) && (shutter.y > 0.0))
+	ccam->shuttertime = shutter.y - shutter.x;
+	
+	const Imath::V2f relativeShutter = shutter - Imath::V2f( Gaffer::Context::current()->getFrame() );
```

Comment:
`const Imath::V2f relativeShutter = Imath::V2f( modf( shutter.x, nullptr ), modf( shutter.y, nullptr ) );`
I'll test here and see.

Reply:
Alright had a bit more of a think to this the next day, modf won't cut it. I'm testing now by bringing this code into Renderer.cpp and hashing the frame number with the camera hash there.

Reply:
Tried out your camerafixes2 branch with the above changes, renders look good to me. 

I guess a simpler way of approaching this would be to calculate the relative shutter time with something like `const Imath::V2f relativeShutter = shutter - Imath::V2f( round(shutter.x + shuttertime / 2.0f) );`